sdv/io/local.py

```diff
@@ -0,0 +1,172 @@
+            The separator used for reading and writing CSV files. Defaults to ``,``.
+        encoding (str):
+            The character encoding to use for reading and writing CSV files. Defaults to ``UTF``.
+        decimal (str):
+            The character used to denote the decimal point. Defaults to ``.``.
+        float_format (str or None):
+            The formatting string for floating-point numbers. Optional.
+
+    Raises:
+        ValueError:
+            If the provided encoding is not available in the system.
+    """
+
+    def __init__(self, sep=',', encoding='UTF', decimal='.', float_format=None):
+        super().__init__(decimal, float_format)
+        try:
+            codecs.lookup(encoding)
+        except LookupError as error:
+            raise ValueError(
+                f"The provided encoding '{encoding}' is not available in your system."
+            ) from error
+
+        self.sep = sep
+        self.encoding = encoding
+
+    def read(self, folder_name, file_names=None):
+        """Read data from CSV files and returns it along with metadata.
+
+        Args:
+            folder_name (str):
+                The name of the folder containing CSV files.
+            file_names (str or list of str, optional):
+                The names of CSV files to read. If None, all files ending with '.csv'
+                in the folder are read.
+
+        Returns:
+            tuple:
+                A tuple containing the data as a dictionary and metadata. The dictionary maps
+                table names to pandas DataFrames. The metadata is an object describing the data.
+
+        Raises:
+            FileNotFoundError:
+                If the specified files do not exist in the folder.
+        """
+        data = {}
+        metadata = MultiTableMetadata()
+
+        folder_path = Path(folder_name)
+
+        if file_names is None:
+            # If file_names is None, read all files in the folder ending with ".csv"
+            file_paths = folder_path.glob('*.csv')
+        else:
+            # Validate if the given files exist in the folder
+            file_names = _cast_to_iterable(file_names)
+            missing_files = [
+                file
+                for file in file_names
+                if not (folder_path / file).exists()
+            ]
+            if missing_files:
+                raise FileNotFoundError(
+                    f"The following files do not exist in the folder: {', '.join(missing_files)}."
+                )
+
+            file_paths = [folder_path / file for file in file_names]
+
+        # Read CSV files
+        for file_path in file_paths:
+            table_name = file_path.stem  # Remove file extension to get table name
+            data[table_name] = pd.read_csv(
+                file_path,
+                sep=self.sep,
+                encoding=self.encoding,
+                parse_dates=False,
+                low_memory=False,
+                decimal=self.decimal
```

Comment:
Actually I did but I'm not sure if this is in the scope of the issue: I was going to suggest adding this:
```
quoting=None,
quotechar='"'
```

Sometimes when exporting people want to add quotation to their data, therefore you find some `csv` with `"John"` instead of `John`. In some cases it could be required, but I assumed that if that was the case, they could export the csvs with the customization that they wanted rather than us handling it with the `CSVHandler`.

Open to suggestions.

Reply:
I'm fine with adding it